File: example/kotlin/src/main/kotlin/com/some/pkg/typesmodule/web/Dtos.kt
package com.some.pkg.typesmodule.web

import com.some.pkg.typesmodule.api.*

data class DateRangeDto(
    val from: String,
    val to: String,
) {
    fun toApi(): DateRange {
        return createDateRange(
            from = createDate(from),
            to = createDate(to),
        )
    }

    companion object {
        fun fromApi(api: DateRange): DateRangeDto {
            return DateRangeDto(
                from = getDateValue(getDateRangeFrom(api)),
                to = getDateValue(getDateRangeTo(api)),
            )
        }
    }
}

File: example/typescript/main/TypesModule/Web/Dtos.ts
namespace TypesModule.Web {
    export class DateRangeDto {
        from = STRING
        to = STRING

        toApi(): DateRange {
            return CustomTypesMapper.dateRangeCreate(
                CustomTypesMapper.dateCreate(this.from),
                CustomTypesMapper.dateCreate(this.to),
            )
        }

        static fromApi(api: DateRange): DateRangeDto {
            const dto = new DateRangeDto()
            dto.from = CustomTypesMapper.dateGetValue(CustomTypesMapper.dateRangeGetFrom(api))
            dto.to = CustomTypesMapper.dateGetValue(CustomTypesMapper.dateRangeGetTo(api))
            return dto
        }
    }
}